starboard/nplb/player_write_sample_test.cc

```diff
@@ -23,103 +25,122 @@ namespace {
+  GroupedSamples samples;
+  if (player_fixture.HasAudio()) {
+    int samples_to_write = SbPlayerGetMaximumNumberOfSamplesPerWrite(
+        player_fixture.GetPlayer(), kSbMediaTypeAudio);
+    samples.AddAudioSamplesWithEOS(0, samples_to_write);
+  }
+  if (player_fixture.HasVideo()) {
+    int samples_to_write = SbPlayerGetMaximumNumberOfSamplesPerWrite(
+        player_fixture.GetPlayer(), kSbMediaTypeVideo);
+    samples.AddVideoSamplesWithEOS(0, samples_to_write);
+  }
 
-void SbPlayerWriteSampleTest::TearDown() {
-  ASSERT_NO_FATAL_FAILURE(player_fixture_.reset());
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
 }
 
-TEST_P(SbPlayerWriteSampleTest, SeekAndDestroy) {
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->Seek(kSbTimeSecond));
-}
+TEST_P(SbPlayerWriteSampleTest, WriteMultipleBatches) {
+  SbPlayerTestFixture player_fixture(GetParam());
+  EXPECT_FALSE(HasFailure());
 
-TEST_P(SbPlayerWriteSampleTest, NoInput) {
-  if (test_media_type_ == kSbMediaTypeAudio) {
-    ASSERT_NO_FATAL_FAILURE(player_fixture_->Write(AudioEOS()));
-  } else {
-    SB_DCHECK(test_media_type_ == kSbMediaTypeVideo);
-    ASSERT_NO_FATAL_FAILURE(player_fixture_->Write(VideoEOS()));
+  int audio_samples_per_batch = 0;
+  audio_samples_per_batch = SbPlayerGetMaximumNumberOfSamplesPerWrite(
+      player_fixture.GetPlayer(), kSbMediaTypeAudio);
+
+  int video_samples_per_batch = 0;
+  video_samples_per_batch = SbPlayerGetMaximumNumberOfSamplesPerWrite(
+      player_fixture.GetPlayer(), kSbMediaTypeVideo);
+
+  GroupedSamples first_batch;
+  if (player_fixture.HasAudio()) {
+    first_batch.AddAudioSamples(0, audio_samples_per_batch);
+  }
+  if (player_fixture.HasVideo()) {
+    first_batch.AddVideoSamples(0, video_samples_per_batch);
   }
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->WaitForPlayerEndOfStream());
-}
 
-TEST_P(SbPlayerWriteSampleTest, WriteSingleBatch) {
-  int max_batch_size = SbPlayerGetMaximumNumberOfSamplesPerWrite(
-      player_fixture_->GetPlayer(), test_media_type_);
-  if (test_media_type_ == kSbMediaTypeAudio) {
-    ASSERT_NO_FATAL_FAILURE(
-        player_fixture_->Write(AudioSamples(0, max_batch_size).WithEOS()));
-  } else {
-    SB_DCHECK(test_media_type_ == kSbMediaTypeVideo);
-    ASSERT_NO_FATAL_FAILURE(
-        player_fixture_->Write(VideoSamples(0, max_batch_size).WithEOS()));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(first_batch));
+
+  GroupedSamples second_batch;
+  if (player_fixture.HasAudio()) {
+    second_batch.AddAudioSamplesWithEOS(audio_samples_per_batch,
+                                        audio_samples_per_batch);
+  }
+  if (player_fixture.HasVideo()) {
+    second_batch.AddVideoSamplesWithEOS(video_samples_per_batch,
+                                        video_samples_per_batch);
   }
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->WaitForPlayerEndOfStream());
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(second_batch));
```

Comment:
We'll eventually resolve the issue where the audio and video streams have significantly different batch size, so one of them won't be able to write through for the second batch, as the SbPlayer is not sending need data.
Feel free to add a TODO so we can work on this later.

Reply:
I changed it to
```
if (HasFatalFailure()) {
  return;
}
```

From gtest doc (http://go/gunitadvanced#checking-for-failures-in-the-current-test), it seems to be a common case.

Reply:
Filed b/283533109 for that.